Rick-and-Morty/Rick And Morty/ViewModels/CharactersViewModel.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+
+final class CharactersViewModel: ObservableObject {
+    let characterType: CharacterType
+    
+    @Published var characters: [Character] = []
+    var title: String = ""
+    var imagePosition: CharacterImagePosition = .right
+    
+    init(characterType: CharacterType) {
+        self.characterType = characterType
+        self.title = getTitle()
+        self.imagePosition = getImagePosition()
+        
+        fetchCharacters()
+    }
+    
+    private func fetchCharacters() {
```

Comment:
This ViewModel has 2 responsibilities: 

- Feed the views with ViewStates
- Decide how to fetch the data

Can you imagine how you can move one of this responsibility out?

Reply:
Moved this out into a CharactersManager class. The class needs to be modified if new CharacterTypes are added, have not been able to find a way around this.